apps/messages/js/bootstrap.js

```diff
@@ -271,6 +271,37 @@ document.addEventListener(
     panelFactory.panelManager = panelManager;
 
     panelManager.processLocation();
+
+    // Listen to messages from the service worker.
+    navigator.serviceWorker.onmessage = async (event) => {
+      let data = event.data;
+      let threads = panelFactory.ui.manager.getThreads().values();
```

Comment:
This is the same code as in https://github.com/capyloon/nutria/blob/a3354e6d4a2c140eab9769a15ac4cbc1d4d832d4/apps/messages/js/bootstrap.js#L51
We should factor that out instead of repeating it since it's more than a few lines.